### ISSUE TITLE
g_ri3Dc fails with 'XDR write failed'

### ISSUE THREAD

**Issue by jsvdvis** [orbeckst/gridcount#6]
Dear all,

Whenever I try to use g_ri3Dc in GROMACS version 4.5.7 with options:
>g_ri3Dc -f hyper_MD.trr -s hyper_MD.tpr -n hyper.ndx -R 13.5 -z1 -15 -z2 15 -b 1000 -e 150000

I get the following error:

> Writing 3D density grid with header
> {T}149100ps{grp}CHAIN{z1}-15.00{z2}15.00{Rmax}13.50{<Delta>}0.050{<N>}6655.11{XTC}hyper_MD.trr{Delta}(0.050,0.050,0.050){sumP}9.92277e+08ps
> 
> grid_write(): XDR write failed.
> Writing the 3D grid failed.

An output file is being created, it contains the header line and some unreadable characters. 
I have also tried several other options with no discernible improvements. It seems something is wrong with XDR, but I have no idea what could be the issue. Could you help me?

Thank you in advance,
Jaap

**Comment by orbeckst** [comment]
Is it always failing or only for one particular example, e.g., did you try a different trajectory, a different selection, different start/end time... 

Does your disk have sufficient free space?

Just from seeing the error message I have no idea why XDR write is failing.

**Comment by orbeckst** [comment]
(Also, because I don't really have the time to support and hack on this tool,  I tend to recommend using other tools such as our MDAnalysis Python library https://github.com/orbeckst/gridcount/issues/4#issuecomment-234800589.)

**Comment by jsvdvis** [comment]
It seems to always fail, I've checked it for several trajectories, index groups and start/end times, as you suggested. I'm also quite sure my system has enough free space (1TB+). 
Unfortunately, I do not know Python well enough to do anything useful with it for now. I'm planning on studying it in about two months, when I'm supposed to have more free time. For now I will work with g_densmap for the density-calculation.

Thank you for your time!

**Comment by orbeckst** [comment]
@jsvdvis thanks for trying out different things. From what you are saying it appears indeed to be a more fundamental problem. 

Did it ever work for you?

**Comment by jsvdvis** [comment]
No, it didn't. It seems to me that the calculating (counting) process is
working just fine, but the writing process seems to fail.

I'm using a newly installed (about a month ago) version of Xubuntu, if that
helps with anything.

Op 21 mei 2017 11:04 schreef "Oliver Beckstein" <notifications@github.com>:

**Comment by orbeckst** [comment]
I haven't compiled gridcount in a while. Maybe something subtle changed in the xdr libraries (although unlikely). 

Did you see any "obvious" warnings when you compiled?

--
Oliver Beckstein
email: orbeckst@gmail.com

**Comment by jsvdvis** [comment]
The only thing I can remember was some error with the libraries (not the
ones you have to give the path to in the Makefile, I can't remember the
exact name), which was solved when I added them to the library path.

2017-05-22 18:04 GMT+02:00 Oliver Beckstein <notifications@github.com>:
